.travis.yml

```
@@ -1,7 +1,12 @@
 language: go
-
 go: 1.14.x
 
+branches:
+  only:
+  - master
+  # Match tags with v<semver>.
+  - /^v(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)(?:-((?:0|[1-9][0-9]*|[0-9]*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9][0-9]*|[0-9]*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$/
```

Hmm, this is [documented](https://docs.travis-ci.com/user/conditions-v1#regular-expressions) to be _Ruby_ regular expression, so we should be able to abuse [options](https://ruby-doc.org/core-2.6/Regexp.html#class-Regexp-label-Options):

```suggestion
  - >-
      /^(?x)
      v(0|[1-9][0-9]*)  # major
      \.
      (0|[1-9][0-9]*)  # minor
      \.
      (0|[1-9][0-9]*)  # patch
      (?:
        - # pre-release
        (
          (?:0|[1-9][0-9]*|[0-9]*[a-zA-Z-][0-9a-zA-Z-]*)
          (?:\.
            (?:0|[1-9][0-9]*|[0-9]*[a-zA-Z-][0-9a-zA-Z-]*)
          )*
        )
      )?
      # build
      (?:\+([0-9a-zA-Z-]+
        (?:\.[0-9a-zA-Z-]+)*
      )
    )?
    $/
```

Okay it probably doesn't need to be split out _that_ detailed, but you get the idea :D At least for the line wrapping.

---

I like it! When I try a YAML parser, it doesn't like it, though. It's not clear where the issue is.

---

The problem is the first line...

---

Hmm, so something like this? GitHub comments isn't the best YAML editor…
```suggestion
  - >-
    /^(?x)
    # v(major).(minor).(patch)
    v(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)
    # -(pre-release)
    (?:-((?:0|[1-9][0-9]*|[0-9]*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9][0-9]*|[0-9]*[a-zA-Z-][0-9a-zA-Z-]*))*))?
    # +(build)
    (?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?
    $/
```

I want to use `(?i)` too, but I wanted to avoid it in case something funny like ı (Turkish dotless I) gets pulled in…

---

I incorporated this in https://github.com/kubernetes-sigs/minibroker/pull/86/commits/23afea54ff9061f4bff7f1bd0b433c3938e88521.
Tested with https://github.com/travis-ci/travis-conditions.